### ISSUE TITLE
Configure Alamofire to accept unsigned SSL certs

### ISSUE THREAD

**Issue by dcunited001** [Alamofire/Alamofire#540]
So i'm referring to [https://stackoverflow.com/questions/30163274/how-to-connect-localhost-with-invalid-certificate-using-alamofire](this code snippet on StackOverflow).  However, when my application makes a request to my API hosted locally, the delegate to handle unsigned certs is never called.  I'm using the Swift 2.0 branch, by the way.

Here's the error I'm seeing in the console.

```
trying to accept invalid certs
2015-06-21 20:11:35.754 App[18885:5060284] Unbalanced calls to begin/end appearance transitions for <UINavigationController: 0x7f8b8b003000>.
https://api.myapp.dev:3000/auth
https://api.myapp.dev:3000/auth
2015-06-21 20:11:38.621 App[18885:5060370] CFNetwork SSLHandshake failed (-9806)
2015-06-21 20:11:38.623 App[18885:5060370] NSURLSession/NSURLConnection HTTP load failed (kCFStreamErrorDomainSSL, -9806)
nil
Optional(Error Domain=NSURLErrorDomain Code=-1200 "An SSL error has occurred and a secure connection to the server cannot be made." UserInfo=0x7f8b8a64c9c0 {_kCFStreamErrorCodeKey=-9806, NSLocalizedRecoverySuggestion=Would you like to connect to the server anyway?, NSUnderlyingError=0x7f8b8a60dd00 "The operation couldn’t be completed. (kCFErrorDomainCFNetwork error -1200.)", NSLocalizedDescription=An SSL error has occurred and a secure connection to the server cannot be made., NSErrorFailingURLKey=https://api.myapp.dev:3000/api/v1/users.json, NSErrorFailingURLStringKey=https://api.myapp.dev:3000/api/v1/users.json, _kCFStreamErrorDomainKey=3})
```

Here's the function I'm calling to set up the delegate for Alamofire.  This is called in my AppDelegate class, in the application() callback.

```
    func acceptInvalidSSLCerts() {
        let manager = Alamofire.Manager.sharedInstance
        print("trying to accept invalid certs")
        
        manager.delegate.sessionDidReceiveChallenge = { session, challenge in
            var disposition: NSURLSessionAuthChallengeDisposition = .PerformDefaultHandling
            var credential: NSURLCredential?
            
            print("received challenge")
            
            if challenge.protectionSpace.authenticationMethod == NSURLAuthenticationMethodServerTrust {
                disposition = NSURLSessionAuthChallengeDisposition.UseCredential
                credential = NSURLCredential(forTrust: challenge.protectionSpace.serverTrust!)
            } else {
                if challenge.previousFailureCount > 0 {
                    disposition = .CancelAuthenticationChallenge
                } else {
                    credential = manager.session.configuration.URLCredentialStorage?.defaultCredentialForProtectionSpace(challenge.protectionSpace)
                    
                    if credential != nil {
                        disposition = .UseCredential
                    }
                }
            }
            
            return (disposition, credential)
        }
    }

```

**Comment by cnoon** [comment]
What happens if you override the `taskDidReceiveChallenge` closure instead? Do you receive the callback then?

**Comment by jshier** [comment]
@dcunited001 `-9806` is also the error you'd receive if you were running into an issue with the new App Transport Security, which is understandable. Perhaps using one of the ATS configuration flags in your `Info.plist` would alleviate the issue?

**Comment by cnoon** [comment]
Thanks @jshier. I'm going to close this issue out @dcunited001. If you provide more info that leads us to believe there is actually an issue, I'll gladly re-open the ticket.

**Comment by dcunited001** [comment]
So instead of following that StackOverflow, I figured out how to correctly configure `Info.plist` to specify exceptions.  I'll go ahead and document this a bit, in case other iOS noobs come across this issue.  

So it's easy to totally turn off `NSAppTransportSecurity`, but there was conflicting information on which keys to use to specify exceptions.  Under the `NSAppTransportSecurity` key, I have `NSAllowsArbitraryLoads` set to `NO`, specifying that I do not want to connect to all HTTP servers.  I also have a `NSExceptionDomains` key set to a Dictionary value, which is where I can specify HTTP domains my app is allowed to connect to.

I edited /etc/hosts to add a custom `api.myapp.dev` domain name.  Then i added a `myapp.dev` key to `Info.plist` under the `NSExceptionDomains` key.  Under the `api.myapp.dev` key, I have set both `NSIncludesSubdomains` and `NSTemporaryExceptionAllowsInsecureHTTPLoads` set to YES.  When I was googling around, I found that `NSTemporaryExceptionAllowsInsecureHTTPLoads` was referred to with two or more different key names, so that was quite confusing to me as an iOS beginner.
